sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasingSuite.scala

```diff
@@ -221,6 +221,42 @@ class NestedColumnAliasingSuite extends SchemaPruningTest {
     comparePlans(optimized, expected)
   }
 
+  test("Do not generate redundant aliases if parent nested field is aliased too") {
+    val nestedRelation = LocalRelation('a.struct('b.struct('c.int,
+      'd.struct('f.int, 'g.int), 'e.int)))
+
+    val first = GetStructField('a, 0, Some("b"))
```

Comment:
can we use expression DSL to write tests?

Reply:
Seems only `Add` have DSL? Don't see DSL for `GetStructField`. I replaced `Add`.

Reply:
Got it. We have `getField`.